### PR TITLE
Compact Mode Setting (Frontend)

### LINKED ISSUE #2891: Show fixed amount of characters per-note (add show more..)
### Describe the solution you'd like

It would be great and nicer if every note is not open by default.

Meaning: if a note is very long, it will be "in the middle" of all the other notes and you would need to scroll down to the end of it.

Having a default number of characters to show per-note could be a nice way of having more "order" in the view, with the possibility to just click on a "show more" button to fully open the note.

### Type of feature

User Interface (UI)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MehadND**
chore: frontend related files for compact view setting.

Possibly close #2891 and #2863.

### CODE REVIEW

**Comment by MehadND** [comment]
### Demo


https://github.com/usememos/memos/assets/34424878/cb2e7327-e007-4ae1-8bdc-fc1411c1233d

**Comment by MehadND** [comment]
Please do note that the expand/shrink content is not persistent, in other words, in compact mode:

if a memo is expanded on home page then if user goes to timeline or profile page then same memo will not be expanded but instead will be shrinked.

**Comment by MehadND** [comment]
@boojack Any chance for you to review this PR? I am certain the backend and DB stuff is already merged, just not the frontend bit. Thank you.

File: web/src/components/MemoView.tsx [hunk 1]
```diff
@@ -138,16 +159,36 @@ const MemoView: React.FC<Props> = (props: Props) => {
...
+          content={memo.content}
+          readonly={readonly}
+          onClick={handleMemoContentClick}
+        />
+        <MemoResourceListView resources={memo.resources} />
+        <MemoRelationListView memo={memo} relations={referenceRelations} />
+        <MemoReactionistView memo={memo} reactions={memo.reactions} />
+      </div>
+      <div className="flex items-center justify-center w-full">
+        {showExpandButton && (
```
Comment (boojack) [inline web/src/components/MemoView.tsx:178]: Let's go to the detail page of memo when clicking `show more` button.

File: web/src/components/MemoView.tsx [hunk 2]
```diff
@@ -85,6 +85,27 @@ const MemoView: React.FC<Props> = (props: Props) => {
...
+      setShowExpandButton(true);
+    } else {
+      setExpand(false);
+      setShowExpandButton(false);
+    }
+  }, [userStore.userSetting?.compactView]);
+
+  const handleCompactModeChanged = () => {
+    if (userStore.userSetting?.compactView) {
+      console.log("Expanding/Shrinking only in compact mode.");
```
Comment (boojack) [inline web/src/components/MemoView.tsx:104]: Please remove debug codes.

**Review by boojack — CHANGES_REQUESTED** [review]
Two more thinks:
1. There should be a linear gradients backgound when compact mode enabled;
2. We'd like to go to memo detail page when clicking the `Show more` button.

image

image

https://tailwindcss.com/docs/background-image#linear-gradients

**Pull request closed without merging**